### PR TITLE
Add support non-numeric signal numbers in RRunProfile

### PR DESCRIPTION
**Pull request by Diadlo**

### CODE REVIEW

File: libr/socket/run.c [hunk 1]
```diff
@@ -355,6 +355,62 @@ R_API int r_run_parsefile (RRunProfile *p, const char *b) {
...
+	{ "SIGWINCH", SIGWINCH },
+	{ "SIGIO", SIGIO },
+	{ "SIGPOLL", SIGPOLL },
+	{ "SIGPWR", SIGPWR },
+	{ "SIGSYS", SIGSYS },
+	{ "SIGSTKSZ", SIGSTKSZ },
+	{ NULL }
+};
+
+static int get_signal (const char *e) {
```
Comment (radare) [inline libr/socket/run.c]: This is already done in libr/debug/signal.c but maybe makes more sense to have it in libr/util/sys.c

Comment (Diadlo) [inline libr/socket/run.c]: Do you talk about `r_debug_signal_resolve`? If yes, it requires `RDebug` and can't be used here.
Should I move this function in `libr/util/sys.c`?

**Comment by radare** [comment]
yep that was the idea, this definition depends on local system, so it doesnt really belongs to debug (it is there because it is the native debug plugin, which depends on the local system), but makes sense to have it in r_util

**Comment by radare** [comment]
ping

**Comment by Diadlo** [comment]
@radare I plan to fix it on this week, but a bit later. Sorry

**Comment by Diadlo** [comment]
@radare I'm not sure, that understood you correctly. But seems, done

File: libr/socket/run.c [hunk 2]
```diff
@@ -405,16 +404,15 @@ R_API int r_run_parseline (RRunProfile *p, char *b) {
 	else if (!strcmp (b, "setgid")) p->_setgid = strdup (e);
 	else if (!strcmp (b, "setegid")) p->_setegid = strdup (e);
 	else if (!strcmp (b, "nice")) p->_nice = atoi (e);
+	else if (!strcmp (b, "timeout")) p->_timeout = atoi (e);
+    else if (!strcmp (b, "timeoutsig")) p->_timeout_sig = r_signal_resolve (e);
```
Comment (radare) [inline libr/socket/run.c]: use tabs

File: libr/include/r_util/r_signal.h [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#ifndef R_SIGNAL_H
+#define R_SIGNAL_H
+
+R_API int r_signal_resolve (const char *str);
```
Comment (radare) [inline libr/include/r_util/r_signal.h]: maybe r_signal_to_string() and r_signal_from_string()?

**Comment by radare** [comment]
We should remove the code in debug and use this api instead

**Comment by radare** [comment]
(to avoid dupes)

**Comment by radare** [comment]
also iirc there was some of those signals that are not supported in all Unixes (some bsd may fail to build). we should test that

**Comment by Diadlo** [comment]
@radare It's ok to use signals from [wiki](https://en.wikipedia.org/wiki/Unix_signal#POSIX_signals)?

**Comment by Diadlo** [comment]
@radare ping :)

**Comment by radare** [comment]
Please rebase

File: libr/util/signal.c [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		if (!strcmp (e, str)) {
+			return signals[i].code;
+		}
+	}
+
+	return atoi (e);
+}
+
+R_API const char* r_signal_to_string (int code) {
+  int i;
```
Comment (radare) [inline libr/util/signal.c]: use tabs

**Comment by radare** [comment]
i'll merge after the use of tabs and rebased on master to see what travis thinks

**Comment by radare** [comment]
tired of waiting. :P

**Pull request merged**